spec.bs

```diff
@@ -1024,41 +1025,101 @@ partial interface Navigator {
-
-
-auction.
+The <dfn for=Navigator method>updateAdInterestGroups()</dfn> method steps are:
 
-<div algorithm="interest group update">
-
-Both update methods trigger the same algorithm, <dfn>interest group update</dfn> which is described below, and takes a list of owners to
-update called "owners":
+1. [=in parallel=] execute the algorithm [=interest group update=] using « [=this=]'s
+  [=relevant settings object=]'s [=environment/top-level origin=] » as `owners`.
+  
+<div algorithm>
+  To <dfn>interest group update</dfn> a given [=list=] of [=url/origins=] |owners|:
 
-1. For each interest group "owner" in "owners":
-  1. Load the interest groups of "owner" where the interest group's "next update after" is before now.
+1. [=list/For each=] |owner| in |owners|:
+  1. [=list/For each=] |ig| of the user agent's [=interest group set=] whose
+    [=interest group/owner=] is |owner| and [=interest group/next update after=] is before now:
     Implementations may consider loading only a portion of these interest groups at a time to avoid
-    issuing too many requests at once. While the loaded set isn't empty:
-    1. For each loaded interest group:
-      1. Fetch the dailyUpdateUrl of the interest group from the network.
-      1. If the network fetch failed, continue to the next interest group.
-      1. Parse the response body as JSON. If parsing fails, continue to the next interest group.
-      1. If the response JSON object is not a JSON dict, continue to the next interest group.
-      1. If "name" and or "owner" keys are present in the response dict, they must match the "name" and "owner" of
-        the original interest group, respectively. If they don't, continue to the next interest group.
-      1. For each remaining field, if the field is one of {priority, enableBiddingSignalsPrioritization,
-        priorityVector, prioritySignalsOverrides, sellerCapabilities, executionMode, biddingLogicUrl,
-        biddingWasmHelperUrl, trustedBiddingSignalsUrl, trustedBiddingSignalsKeys, ads,
-        adComponents, adSizes, sizeGroups}:
-        1. If the field is prioritySignalsOverrides, merge with the stored value: for each key, value pair in the JSON prioritySignalsOverrides:
-          1. If the value is null, delete the stored prioritySignalsOverrides key, value pair.
-          2. Otherise, update the stored value for key to be the new value.
-        1. Otherwise, replace the current value of the field in the interest group with that field from the received
-          interest group update JSON. (Fields that are not specified in the update JSON will remain unmodified).
-        1. Set the "last updated" of the interest group to the current time, and set the "next update after" to
-          the update rate limit period + the current time.
+    issuing too many requests at once.
+    1. Let |update| be the result of [=fetching resource=] with |ig|'s [=interest group/daily update url=]

```

Comment:
Just want to make sure: per the current spec, this request will be credential-less, `no-cors`, and with an opaque origin; does that match the implementation?

Reply:
> credential-less,

Yes: https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/interest_group_update_manager.cc;l=603;drc=9329f990b89052f1e7f82e0a1a4b298b72359263;bpv=1;bpt=1

> no-cors, 

Yes, this is the default: https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:services/network/public/cpp/resource_request.h;l=150;drc=9329f990b89052f1e7f82e0a1a4b298b72359263

> with an opaque origin

Hmm, looks like we set it to the IG owner: https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/browser/interest_group/interest_group_update_manager.cc;l=604;drc=9329f990b89052f1e7f82e0a1a4b298b72359263;bpv=1;bpt=1
which makes sense as that's the origin that added the IG and requested it's update.  Let me see about modifying the spec to match.

Reply:
In general it’s starting to look like we might need an overhaul of how fetching is done in this spec. Things seemed centralized about 2 algorithms but I feel like we keep seeing instances where that is insufficient for the various unique fetches in this spec.

Reply:
I fixed the origin for the update fetch.  I'm open to other ways to rework the fetching.